### PR TITLE
chore(screenshotter): Improve screenshotter for Safari

### PR DESCRIPTION
**Pull request by edemaine**
This PR does two things, and fixes or improves #3012:

1. ~~Upgrade to Safari 14.0; seems good to track modern Safari~~ I accidentally did this only locally, not in CI. I reverted this change because I didn't want to deal with changed screenshots in this PR.
2. Fix running Docker on IPv6 system. This is an aside I guess, but was necessary in my Windows environment; unfortunately, I still couldn't get the screenshotter working on Windows; it seems to just hang while trying to connect to the Selenium docker.
3. Ensure images load before screenshotting, the same way we do with fonts. This is presumably what was causing the `Includegraphics` test to fail on Safari (dunno why it works on other browsers; maybe they load local files fast).

CI doesn't crash on `Includegraphics` tests anymore.  I've seen Safari crash with a "failure to find function `load_fonts_and_images`" error and a timeout error.  I still haven't seen Chrome or Firefox crash, so this PR doesn't seem to have broken anything. I think Safari tests succeed more often now but there might be work left to do.

### CODE REVIEW

File: test/screenshotter/test.html [hunk 1]
```diff
@@ -31,8 +31,8 @@
     <span id="math"></span>
     <span id="post"></span>
     <script type="text/javascript">
-      function load_fonts(callback) {
-        if (!document.fonts || !document.fonts.load) {
+      function load_fonts_and_images(callback) {
+        if (!document.fonts || !document.fonts.load || !document.images || !document.images.load) {
```
Comment (ylemkimon) [inline test/screenshotter/test.html:35]:
```suggestion
        if (!document.fonts || !document.fonts.load) {
```
`document.images` is available on all browsers and there is no `document.images.load`.

File: test/screenshotter/test.html [hunk 2]
```diff
@@ -56,6 +56,17 @@
         load_font("Mincho");
         load_font("Batang");
 
+        document.images.forEach(function (img) {
```
Comment (ylemkimon) [inline test/screenshotter/test.html:59]: AFAIK, there is no `forEach` method on `document.images` (`HTMLCollection`).

**Comment by edemaine** [comment]
@ylemkimon Thanks for spotting those bugs which caused the code to never run! (and broke font loading) I tweaked the code to always do the image loading checks, correctly this time, and no longer break font loading.

I've run the tests three times, and while Safari doesn't always get the Includegraphics test in the first try, it so far hasn't actually failed within the 5 attempts. Anyway, I think this PR should help and can't hurt, even if we remove the test for Safari.

File: test/screenshotter/test.html [hunk 3]
```diff
@@ -31,14 +31,11 @@
...
     <span id="post"></span>
     <script type="text/javascript">
-      function load_fonts(callback) {
-        if (!document.fonts || !document.fonts.load) {
-          callback();
-          return;
-        }
-
+      function load_fonts_and_images(callback) {
         var results = [];
```
Comment (ylemkimon) [inline test/screenshotter/test.html:35]:
```suggestion
        if (typeof Promise === "undefined") {
          callback();
          return;
        }

        var results = [];
```

**Review by ylemkimon — APPROVED** [review]

**Comment by ylemkimon** [comment]
Thank you for the PR! The Safari screenshotter always bothered me.

**Comment by edemaine** [comment]
@ylemkimon Thanks for the review and improvement!

**Pull request merged**